src/OptionContainer.cpp

```diff
@@ -75,240 +63,325 @@ void OptionContainer::deletePlugins(std::deque<Plugin *> &list)
+                    temp = temp.before("#");
+                }
+                temp.removeWhiteSpace(); // get rid of spaces at end of line
+                while (temp.contains("__LISTDIR__")) {
+                    String temp2 = temp.before("__LISTDIR__");
+                    temp2 += now_pwd;
+                    temp2 += temp.after("__LISTDIR__");
+                    temp = temp2;
+                }
+                // deal with included files
+                if (temp.startsWith(".")) {
+                    temp = temp.after(".Include<").before(">");
+                    if (temp.length() > 0) {
+                        temp.fullPath(base_dir);
+                        if (!readConfFile(temp.toCharArray(), now_pwd)) {
+                            conffiles.close();
+                            return false;
                         }
-                    } else 	if ((log_location = findoptionS("loglocation")) == "") {
-                        log_location = __LOGLOCATION;
-                        log_location += "/access.log";
-                        log_syslog = false;
                     }
-#ifndef NEWDEBUG_OFF
-		    if((debuglevel = findoptionS("debuglevel")) != "")
-		    {
+                    String temp2 = temp.after(".Define LISTDIR <").before(">");
+                    if (temp2.length() > 0) {
+                        now_pwd = temp2;
+                        //if(!now_pwd.endsWith("/"))
+                        //    now_pwd += "/";
+                    }
+                    continue;
+                }
+                // append ,listdir=now_pwd if line contains a file path - so that now_pwd can be passed
+                // to list file handler so that it can honour __LISTDIR__ in Included listfiles
+                if (temp.contains("path=") && !temp.contains("listdir=")) {
+                    temp += ",listdir=";
+                    temp += now_pwd;
+                }
 
-		    }
-		    if((path_debuglevel = findoptionS("debuglevelfile")) != "")
-		    {
+                // if (temp.startsWith(LoggerConfigurator::Prefix))
+                //     loggerConf.configure(temp);
 
-		    }
-		    myDebug = new DebugManager(debuglevel, path_debuglevel);
-#endif
-                    if ((RQlog_location = findoptionS("rqloglocation")) == "") {
-                        log_requests = false;
-                    } else {
-                        log_requests = true;
-                    }
+                linebuffer = temp.toCharArray();
+                DEBUG_config("read:", linebuffer);
+                conffile.push_back(linebuffer); // stick option in deque
+            }
+        }
+    }
+    conffiles.close();
+    return true;
+}
 
-                    if ((stat_location = findoptionS("statlocation")) == "") {
-                        stat_location = __LOGLOCATION;
-                        stat_location += "/stats";
-                    }
+bool OptionContainer::read(std::string &filename, int type) {
```

Comment:
Codacy found an issue: [Check buffer boundaries if used in a loop including recursive loops (CWE-120, CWE-20).  ](https://app.codacy.com/manual/numsys/e2guardian/pullRequest?prid=6238852)